torch/_functorch/aot_autograd.py

```diff
@@ -918,10 +918,27 @@ def aot_module_simplified(
             aot_config,
         )
 
+    if isinstance(mod, torch._dynamo.utils.GmWrapper):
+        # When we have list inputs to the graph, this function is called by the flatten_graph_inputs
+        # wrapper, which boxes the inputs so that they can be freed before the end of this scope
+        def boxed_forward(runtime_args: List[torch.Tensor]):
```

Comment:
There's probably a good reason for this, but is there a reason we can't use this boxed_forward by default? (and require dynamo to **always** pass its graph inputs in boxed form)? It feels nicer not to have a major divergence in how the calling convention in AOTAutograd works, which is already pretty complicated

Reply:
We can have it as default only by paying the runtime overhead of converting the inputs from a `Tuple[Any]` to `List[Any]`. This is because dynamo OptimizedModule is a nn.Module, and nn.Module doesn't box, so the conversion is required for any user program that accepts non-list inputs

If we decide to go ahead with that, although it would be cleaner within this function:
```python
def call_with_list_args(*args):  # Note: holds refs to each arg
    return boxed_forward(list(args))
```
we would need to deal with refcount in the `flatten_graph_inputs` wrapper, perhaps by wrapping the flattened inputs in a nested list to survive the unpacking e.g.
```python
def flatten_graph_inputs(gm: torch.fx.GraphModule, inputs, compile_gm):
    ...
    def wrapper(*args):
        flat_args: List[Tensor] = pytree.arg_tree_leaves(*args)
        return compiled_fn([flat_args])  # extra list to pass through call_with_list_args's unpacking without holding a ref
    return wrapper
```

Reply:
Ah this makes sense. Thanks! Can you mention that context in the comment?